vm/proxyapp/proxyappclient.go

```diff
@@ -97,6 +92,22 @@ func (p *pool) init(params *proxyAppParams, cfg *Config) error {
 	return nil
 }
 
+func (p *pool) closeProxy() {
+	if p.proxy != nil {
+		if p.proxy.Client != nil {
```

Comment:
Is it OK to close the client concurrently with ProxyVM.PoolLogs calls?
For the production stdInOutCloser codec it seems to be OK, since Close method is nop. Not sure about the testing implementation. Can't it produce some spurious error messages on races, or even crash?
A cleaner approach would be to join the logging goroutine first (signal to exit and wait for its exit) and then close the client when nobody is using it.

Reply:
Let's go right way. Added 1 more channel.